testkit/shared/src/main/scala/org/typelevel/otel4s/testkit/Metric.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.testkit
+
+final case class Metric(
```

Comment:
I'm not sure about the `case class` all over the testkit module. It's a nightmare in terms of bincompat.  

On the other hand, it comparison works out of the box.

Reply:
Since we have Cats here, you an implement an `Eq` (`Hash`, `Order`?) instance by converting to tuple `Eq.by(metric => (metric.name, metric.description ...))`.

Then, you can implement `equals` in terms of `eq`.

Reply:
That sounds good. 

I assume we can add `Show` as well? Some test frameworks rely on the text representation of data.